assets/src/edit-story/utils/generatePatternCSS.js

```diff
@@ -0,0 +1,124 @@
+    : '';
+  const centerString = center
+    ? ` at ${round(100 * center.x, 2)}% ${round(100 * center.y, 2)}%`
+    : '';
+  const rotationString = rotation ? `${rotation}turn` : '';
+  switch (type) {
+    case 'radial':
+      if (!centerString && !sizeString) {
+        return null;
+      }
+      return `${sizeString}${centerString}`.trim();
+
+    case 'conic':
+      if (!rotationString && !centerString) {
+        return null;
+      }
+      const fromRotationString = rotationString ? `from ${rotationString}` : '';
+      return `${fromRotationString}${centerString}`.trim();
+
+    case 'linear':
+      return rotationString;
+
+    default:
+      return null;
+  }
+}
+
+function getStopList(stops, isAngular = false) {
+  const getPosition = (val) =>
+    isAngular ? `${round(val, 4)}turn` : `${round(val * 100, 2)}%`;
+  const getColor = ({ r, g, b, a = 1 }) => rgba(r, g, b, a);
+  return stops.map(
+    ({ color, position }) => `${getColor(color)} ${getPosition(position)}`
+  );
+}
+
+/**
+ * Generate CSS from a Pattern.
+ *
+ * @param {Object} pattern Patterns as describe by the Pattern type
+ * @param {Object} options Optional settings
+ * @param {string} options.property Type of CSS to generate. Defaults to 'background',
+ * but can also be 'color', 'fill' or 'stroke'.
+ * @param {boolean} options.asString If true (default) generates a string, if false a style object,
+ *
+ * @return {string | Object} CSS declaration as string or object, e.g. 'fill: transparent' or
+ * {backgroundImage: 'radial-gradient(red, blue)'}.
+ */
+function generatePatternCSS(
+  pattern = null,
+  { property = 'background', asString = true } = {}
+) {
+  const isBackground = property === 'background';
+  if (pattern === null) {
+    if (asString) {
+      return `${property}: transparent`;
+    }
+    return { [property]: 'transparent' };
+  }
+
+  const { type = 'solid' } = pattern;
+  if (!['solid', 'radial', 'linear', 'conic'].includes(type)) {
+    throw new Error(`Unknown pattern type: '${type}'`);
+  }
+  // Gradients are only possible for backgrounds
+  if (!isBackground && type !== 'solid') {
+    throw new Error(
+      `Can only generate solid colors for property '${property}'`
+    );
+  }
+
+  if (type === 'solid') {
+    const {
+      color: { r, g, b, a = 1 },
+    } = pattern;
+    if (asString) {
+      const stringPropertyPostfix = isBackground ? '-color' : '';
+      return `${property}${stringPropertyPostfix}: ${rgba(r, g, b, a)}`;
+    }
+    const objectPropertyPostfix = isBackground ? 'Color' : '';
+    return { [`${property}${objectPropertyPostfix}`]: rgba(r, g, b, a) };
+  }
+
+  const { stops } = pattern;
+  const func = `${type}-gradient`;
+  const description = getGradientDescription(pattern);
+  const stopList = getStopList(stops, type === 'conic');
+  const parms = description ? [description, ...stopList] : stopList;
+  const value = `${func}(${parms.join(', ')})`;
+  if (asString) {
+    return `background-image: ${value}`;
```

Comment:
Why note `${property}-image`? I don't think we have borders yet, but this could support `border-image` very similarly.

Reply:
I did not consider border's at all. Do we need it?

Reply:
Checked with UX and border is definitely not in P0. The text gradient, however, seems to be in P0. At least for now. But, certainly, ok to follow up in another PR. Just heads up.

Reply:
Ok. Clarified this as well. No text color gradient in P0 (and maybe not ever). I.e. when we show color picker for text color, the gradient options should be hidden.